cmd/zoekt-sourcegraph-indexserver/cleanup.go

```diff
@@ -98,19 +109,25 @@ func cleanup(indexDir string, repos []uint32, now time.Time, shardMerging bool)
 	}
 
 	// index: Move missing repos from trash into index
+	// index: Restore deleted or tombstoned repos.
 	for _, repo := range repos {
 		// Delete from index so that index will only contain shards to be
 		// trashed.
 		delete(index, repo)
 
-		shards, ok := trash[repo]
-		if !ok {
-			continue
+		if shards, ok := trash[repo]; ok {
+			log.Printf("restoring shards from trash for %v", repo)
+			moveAll(indexDir, shards)
+			shardsLog(indexDir, "restore", shards)
```

Comment:
should we continue here to skip the tombstone codepath? IE it seems possible to have a repo both in trash and tombstoned? I see above you keep the invariant that repo is only in one or the other, but still feels good to have the continue since that code is a little further away and/or document that invariant here.